swift_x_account_sharing/server.py

```diff
@@ -30,32 +34,45 @@
 asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
 
 
-def resume_on_start(app):
+async def kill(seconds, value):
+    """Kill the application after an elapsed time."""
+    await asyncio.sleep(seconds)
+    sys.exit(value)
+
+
+async def resume_on_start(app):
     """Resume old instance from start."""
     # If using dict_db read the database on disk, if it exists
     if (
             isinstance(app["db_conn"], InMemDB)
             and os.path.exists("swift-x-account-sharing.inmemdb")
     ):
-        app["db_conn"].load_from_file("swift-x-account-sharing.inmemdb")
+        await app["db_conn"].load_from_file("swift-x-account-sharing.inmemdb")
+    if isinstance(app["db_conn"], DBConn):
+        await app["db_conn"].open()
 
 
 async def save_on_shutdown(app):
     """Flush the database on shutdown."""
     # If using dict_db dump the database on disk, using default file.
     if isinstance(app["db_conn"], InMemDB):
-        app["db_conn"].export_to_file("swift-x-account-sharing.inmemdb")
+        await app["db_conn"].export_to_file("swift-x-account-sharing.inmemdb")
+    if isinstance(app["db_conn"], DBConn):
+        await app["db_conn"].close()
 
 
 async def init_server():
     """Initialize the server."""
     app = aiohttp.web.Application(
-        middlewares=[add_cors]
+        middlewares=[add_cors, check_db_conn]
     )
 
-    app["db_conn"] = InMemDB()
+    if os.environ.get("SHARING_DB_POSTGRES", None):
+        app["db_conn"] = DBConn()
+    else:
+        app["db_conn"] = InMemDB()
 
-    resume_on_start(app)
+    asyncio.ensure_future(resume_on_start(app))
```

Comment:
should be `on_startup` from https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.Application.on_startup